lib/msf/core/exploit/remote/http/http_cookie_jar.rb

```diff
@@ -80,7 +84,11 @@ def parse(set_cookie_header)
                 next
               end
 
-              cookies << HttpCookie.new(name, value, attrs)
+              if attrs.is_a?(Hash)
+                cookies << HttpCookie.new(name, value, **attrs)
+              else
+                cookies << HttpCookie.new(name, value)
+              end
```

Comment:
What else can `attrs` be? 👀 

```suggestion
              cookies << HttpCookie.new(name, value, **attrs)
```

Reply:
`scan_set_cookie` can return`nil` in place of a hash as `attrs`, which will cause a `wrong number of arguments (given 3, expected 1..2) (ArgumentError)` error.  

I could replace the `if attrs.is_a?(Hash)` with `if attrs`, but some non hash value could be passed in the future if `scan_set_cookie` is changed or if another method entirely is used, which would also cause the above error, so IMO `is_a?(Hash)` is clearier since it indicates what it is we're looking for to ensure no errors.

Reply:
How about
```
attrs ||= {}
cookies << HttpCookie.new(name, value, **attrs)
```

Reply:
That still has the issue of attrs being a value that isn't nil but also isn't a hash

Reply:
Just checking the underlying code, I don't think a non-hash is yielded? 👀 

https://github.com/sparklemotion/http-cookie/blob/2578ebb536c4617fd3b343fb2468d1a3c5b705b8/lib/http/cookie/scanner.rb#L171

Reply:
```suggestion
              raise ArgumentError, 'Expected attrs to be a Hash' unless attrs.is_a?(Hash)
              cookies << HttpCookie.new(name, value, **attrs)
```

Reply:
We'll want tests for this scenario too then

```ruby
  describe '#parse' do
    it 'parses comma separated cookies' do
      input = "cross-site-cookie=bar; SameSite=Strict;, Cacti=qlf907tneuneajqbr00fge0ull; path=/cacti/; HttpOnly"
      expected = [
        # ...
      ]
      expect(subject.parse(input)).to eq(expected)
    end

    it 'validates that that the third party scan_set_cookie implementation always returns a hash' do
      allow_any_instance_of(::HTTP::Cookie::Scanner).to receive(:scan_set_cookie).and_yield "foo", "bar", nil
      input = "cross-site-cookie=bar; SameSite=Strict;, Cacti=qlf907tneuneajqbr00fge0ull; path=/cacti/; HttpOnly"
      expect { subject.parse(input) }.to raise_error /Expected attrs to be a Hash/
    end
  end
```